token/confidential-transfer/elgamal-registry/src/instruction.rs

```diff
@@ -0,0 +1,196 @@
+use {
+    crate::{get_elgamal_registry_address, id},
+    solana_program::{
+        instruction::{AccountMeta, Instruction},
+        program_error::ProgramError,
+        pubkey::Pubkey,
+        system_program, sysvar,
+    },
+    solana_zk_sdk::zk_elgamal_proof_program::{
+        instruction::ProofInstruction, proof_data::PubkeyValidityProofData,
+    },
+    spl_token_confidential_transfer_proof_extraction::{ProofData, ProofLocation},
+};
+
+#[derive(Clone, Debug, PartialEq)]
+#[repr(u8)]
+pub enum RegistryInstruction {
+    /// Initialize an ElGamal public key registry.
+    ///
+    /// 0. `[writeable, signer]` The funding account (must be a system account)
```

Comment:
micro-nit typo
```suggestion
    /// 0. `[writable, signer]` The funding account (must be a system account)
```

Also, we've been trying to avoid having payer accounts passed to the program, and instead assume that there's enough lamports to cover rent-exemption already. This way, the program just needs to allocate + assign.